### ISSUE TITLE
XMPPFramework crash in iOS 16 and Xcode 14

### ISSUE THREAD

**Issue by zhangylios** [robbiehanson/XMPPFramework#1223]
hello, XMPPFramework runs iOS16, crashes on Xcode14 device, the error is like a screenshot, please help to update and solve it, thank you very much, XMPPFramework is great！
<img width="1054" alt="WeChate999a946a8c4024d342e74da6d1e6308" src="https://user-images.githubusercontent.com/21352569/192216294-6e23ba31-03af-43a0-8a4a-0c84edffb2b8.png">

**Comment by andrea8000** [comment]
same problem
@zhangylios did you find any solution?

**Comment by Monks-Giorgio** [comment]
Hi,
check if you have XMPPStream.enableBackgroundingOnSocket set to YES. In case, try not to use it

**Comment by pengxu1013** [comment]
> Hi,
> check if you have XMPPStream.enableBackgroundingOnSocket set to YES. In case, try not to use it

Remove the xmppStream. EnableBackgroundingOnSocket = YES; It solves the crash problem, but does not setting this property affect the functionality?

**Comment by Monks-Giorgio** [comment]
> > Hi,
> > check if you have XMPPStream.enableBackgroundingOnSocket set to YES. In case, try not to use it
> 
> Remove the xmppStream. EnableBackgroundingOnSocket = YES; It solves the crash problem, but does not setting this property affect the functionality?

yep, for sure it affects functionality, if you need it you should use PushKit as documented in the code (from XMPPStream.h#273):

`@property (readwrite, assign) BOOL enableBackgroundingOnSocket DEPRECATED_MSG_ATTRIBUTE("Background sockets are no longer available on iOS 10. You must use PushKit and the XEP-0357 module instead.");`

Also check Apple documentation at [OptimizeVoIP](https://developer.apple.com/library/archive/documentation/Performance/Conceptual/EnergyGuide-iOS/OptimizeVoIP.html)

**Comment by zhangylios** [comment]
> > > Hi,
> > > check if you have XMPPStream.enableBackgroundingOnSocket set to YES. In case, try not to use it
> > 
> > 
> > Remove the xmppStream. EnableBackgroundingOnSocket = YES; It solves the crash problem, but does not setting this property affect the functionality?
> 
> yep, for sure it affects functionality, if you need it you should use PushKit as documented in the code (from XMPPStream.h#273):
> 
> `@property (readwrite, assign) BOOL enableBackgroundingOnSocket DEPRECATED_MSG_ATTRIBUTE("Background sockets are no longer available on iOS 10. You must use PushKit and the XEP-0357 module instead.");`
> 
> Also check Apple documentation at [OptimizeVoIP](https://developer.apple.com/library/archive/documentation/Performance/Conceptual/EnergyGuide-iOS/OptimizeVoIP.html)

This is the answer and solved my problem, thanks a lot

**Comment by pengxu1013** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by mahesh-reddy-ios** [comment]
> > > > Hi,
> > > > check if you have XMPPStream.enableBackgroundingOnSocket set to YES. In case, try not to use it
> > > 
> > > 
> > > Remove the xmppStream. EnableBackgroundingOnSocket = YES; It solves the crash problem, but does not setting this property affect the functionality?
> > 
> > 
> > yep, for sure it affects functionality, if you need it you should use PushKit as documented in the code (from XMPPStream.h#273):
> > `@property (readwrite, assign) BOOL enableBackgroundingOnSocket DEPRECATED_MSG_ATTRIBUTE("Background sockets are no longer available on iOS 10. You must use PushKit and the XEP-0357 module instead.");`
> > Also check Apple documentation at [OptimizeVoIP](https://developer.apple.com/library/archive/documentation/Performance/Conceptual/EnergyGuide-iOS/OptimizeVoIP.html)
> 
> This is the answer and solved my problem, thanks a lot

Can u please post the solution which solve your problem

**Comment by Rajai-Sunnyclock** [comment]
@mahesh-reddy-ios  Check this: https://github.com/robbiehanson/CocoaAsyncSocket/pull/717/commits

**Comment by love-cuteQian** [comment]
> 您好， 请检查您是否将 XMPPStream.enableBackgroundingOnSocket 设置为 YES。以防万一，尽量不要使用

The simulator can be retained. Use this solution directly for real machine debugging.

**Comment by pengxu1013** [comment]
我已经解决了，谢谢




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "robbiehanson/XMPPFramework"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年1月16日(星期二) 下午4:30
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [robbiehanson/XMPPFramework] XMPPFramework crash in iOS 16 and Xcode 14 (Issue #1223)





  
您好， 请检查您是否将 XMPPStream.enableBackgroundingOnSocket 设置为 YES。以防万一，尽量不要使用
  
The simulator can be retained. Use this solution directly for real machine debugging.
